CI flake: Production Docker (flex) sometimes fails with "application not healthy after 10m0s"

## What

The flex production docker job (`build / Production Docker (flex)`) intermittently fails with:

```
application not healthy after 10m0s
##[error]Process completed with exit code 1.
```

Container logs show mysql healthy and openemr started, but the openemr container's healthcheck never passes within the 600-second wait window.

## Where it comes from

`.github/actions/test-actions-core/action.yml` runs:

```yaml
- name: Run the containers
  run: docker compose up --detach --wait --wait-timeout 600 mysql "\${OPENEMR_SERVICE_NAME}"
```

`docker compose up --wait` polls the service healthcheck and exits non-zero with that message when the timeout elapses.

## Recent occurrence

Run https://github.com/openemr/openemr-devops/actions/runs/24732488650 on master (commit 959f246, the merge of #660). Re-running the failed job alone passed without changes, confirming flake.

- #660 only touched `docker/openemr/8.0.0/*` — did not modify the flex container.
- The flex job passed on the PR CI for #660 and on the master push CI for the merge immediately prior (#655, commit 84ca0c3).

## Why it likely flakes

The flex image runs `composer install` at container start. Under CI runner load (cold cache, slow mirror, contention), that can push total boot past 10 min. The wait-timeout is a budget, not a correctness check.

## Suggested mitigations (pick one)

- **Raise the `--wait-timeout`.** 20m would absorb most composer-install variance while still catching real hangs.
- **Warm the image.** Move composer install into the image build (cached layer) rather than container startup, so healthcheck measures runtime boot only.
- **Retry the failed step once** before failing the job. Cheapest fix, doesn't address root cause but silences the flake.
- **Split the healthcheck into cheap (apache up) + slow (installer done) signals**, so compose's `--wait` succeeds quickly while a separate step verifies post-install state.

## Priority

Low — flake is rare and resolved by a single re-run. Open to document the pattern and the mitigation options so the next person who hits it doesn't have to re-derive the diagnosis.

--- a/docker/openemr/flex/openemr.sh
+++ b/docker/openemr/flex/openemr.sh
@@ -288,6 +288,7 @@ is_configured() {
 
 AUTHORITY=yes
 OPERATOR=yes
+SWARM_WAIT_DEFERRED=no
 
 # Kubernetes-specific role assignment
 if [[ "${K8S}" = "admin" ]]; then
@@ -377,17 +378,8 @@ update_leader_heartbeat() {
     fi
 }
 
-# Handles swarm mode coordination: leader election and follower waiting.
-handle_swarm_mode() {
-    # Skip coordination if swarm mode isn't enabled
-    if [[ "${SWARM_MODE}" != "yes" ]]; then
-        return 0
-    fi
-
-    # Try to become the leader
-    try_become_leader
-
-    # If we're a follower, wait for the leader to finish setup
+# Waits for the swarm leader to finish setup before reading shared configuration.
+wait_for_swarm_completion() {
     if [[ "${AUTHORITY}" = "no" && ! -f "${OE_ROOT}/sites/docker-completed" ]]; then
         echo "Waiting for docker-leader to finish configuration (with timeout-based recovery)..."
         local -i max_wait_time="${LEADER_WAIT_TIMEOUT:-600}"  # Default: 10 minutes
@@ -400,8 +392,8 @@ handle_swarm_mode() {
             # shellcheck disable=SC2310  # set -e behavior in conditionals is intentional
             if is_leader_stale; then
                 echo "Leader appears to have crashed, attempting to take over..."
-                # shellcheck disable=SC2310  # set -e behavior in conditionals is intentional
-                if try_become_leader; then
+                try_become_leader
+                if [[ "${AUTHORITY}" = "yes" ]]; then
                     echo "Successfully became leader after previous leader failure"
                     break
                 fi
@@ -412,8 +404,11 @@ handle_swarm_mode() {
 
         # Try one more time to become leader (in case leader died just as we timed out)
         # shellcheck disable=SC2310  # set -e behavior in conditionals is intentional
-        if [[ ! -f "${OE_ROOT}/sites/docker-completed" ]] && try_become_leader; then
-            echo "Promoted to leader after waiting period"
+        if [[ ! -f "${OE_ROOT}/sites/docker-completed" ]]; then
+            try_become_leader
+            if [[ "${AUTHORITY}" = "yes" ]]; then
+                echo "Promoted to leader after waiting period"
+            fi
         fi
 
         # If we timed out, check if configuration actually exists
@@ -427,8 +422,10 @@ handle_swarm_mode() {
             fi
         fi
     fi
+}
 
-    # If we're the leader, create initiation marker and send heartbeat
+# If we're the leader, create initiation marker and restore swarm pieces.
+prepare_swarm_leader() {
     if [[ "${AUTHORITY}" = "yes" ]]; then
         touch "${OE_ROOT}/sites/docker-initiated"
         update_leader_heartbeat
@@ -443,6 +440,28 @@ handle_swarm_mode() {
             rsync --owner --group --perms --recursive --links /swarm-pieces/sites "${OE_ROOT}/"
         fi
     fi
+}
+
+# Handles swarm mode coordination: leader election and follower waiting.
+handle_swarm_mode() {
+    # Skip coordination if swarm mode isn't enabled
+    if [[ "${SWARM_MODE}" != "yes" ]]; then
+        return 0
+    fi
+
+    # Try to become the leader
+    try_become_leader
+
+    # Flex followers can prepare local source/dependencies while the leader
+    # configures the shared sites volume, then wait before reading sqlconf.php.
+    if [[ "${AUTHORITY}" = "no" && ! -f "${OE_ROOT}/sites/docker-completed" ]]; then
+        SWARM_WAIT_DEFERRED=yes
+        echo "Deferring docker-leader wait until local flex build is ready"
+        return 0
+    fi
+
+    wait_for_swarm_completion
+    prepare_swarm_leader
 }
 
 # ============================================================================
@@ -685,6 +704,12 @@ if [[ "${NEED_COMPOSER_BUILD}" = "true" ]] || [[ "${NEED_NPM_BUILD}" = "true" ]]
     fi
 
     cd /var/www/localhost/htdocs
+fi
+
+if [[ "${SWARM_WAIT_DEFERRED}" = "yes" ]]; then
+    wait_for_swarm_completion
+    SWARM_WAIT_DEFERRED=no
+    prepare_swarm_leader
 fi
 
 if [[ "${AUTHORITY}" = "yes" ]] ||

--- a/tests/bats/flex/openemr.bats
+++ b/tests/bats/flex/openemr.bats
@@ -7,6 +7,10 @@ setup() {
     [[ -n "$SCRIPT_DIR" ]] && [[ -d "$SCRIPT_DIR" ]]
 }
 
+@test "flex openemr.sh: valid bash syntax" {
+    assert_script_syntax "${SCRIPT_DIR}/openemr.sh"
+}
+
 @test "flex openemr.sh: FLEX_REPOSITORY and FLEX_REPOSITORY_BRANCH" {
     assert_script_contains "${SCRIPT_DIR}/openemr.sh" 'FLEX_REPOSITORY'
     assert_script_contains "${SCRIPT_DIR}/openemr.sh" 'FLEX_REPOSITORY_BRANCH'
@@ -33,6 +37,12 @@ setup() {
 @test "flex openemr.sh: MANUAL_SETUP, K8S, SWARM_MODE" {
     assert_script_contains "${SCRIPT_DIR}/openemr.sh" 'MANUAL_SETUP='
     assert_script_contains "${SCRIPT_DIR}/openemr.sh" 'SWARM_MODE='
+}
+
+@test "flex openemr.sh: swarm follower defers leader wait for flex build" {
+    assert_script_contains "${SCRIPT_DIR}/openemr.sh" 'SWARM_WAIT_DEFERRED=yes'
+    assert_script_contains "${SCRIPT_DIR}/openemr.sh" 'Deferring docker-leader wait until local flex build is ready'
+    assert_script_contains "${SCRIPT_DIR}/openemr.sh" 'wait_for_swarm_completion'
 }
 
 @test "flex openemr.sh: INSANE_DEV_MODE or DEVELOPER_TOOLS" {

